packages/browser-integration-tests/standalone-custom-key.html

```diff
@@ -0,0 +1,97 @@
+        if (!analytics.initialize)
+          if (analytics.invoked)
+            window.console &&
+              console.error &&
+              console.error('Segment snippet included twice.')
+          else {
+            analytics.invoked = !0
+            analytics.methods = [
+              'trackSubmit',
+              'trackClick',
+              'trackLink',
+              'trackForm',
+              'pageview',
+              'identify',
+              'reset',
+              'group',
+              'track',
+              'ready',
+              'alias',
+              'debug',
+              'page',
+              'screen',
+              'once',
+              'off',
+              'on',
+              'addSourceMiddleware',
+              'addIntegrationMiddleware',
+              'setAnonymousId',
+              'addDestinationMiddleware',
+              'register',
+            ]
+            analytics.factory = function (e) {
+              return function () {
+                if (window[i].initialized)
+                  return window[i][e].apply(window[i], arguments)
+                var n = Array.prototype.slice.call(arguments)
+                if (
+                  [
+                    'track',
+                    'screen',
+                    'alias',
+                    'group',
+                    'page',
+                    'identify',
+                  ].indexOf(e) > -1
+                ) {
+                  var c = document.querySelector("link[rel='canonical']")
+                  n.push({
+                    __t: 'bpc',
+                    c: (c && c.getAttribute('href')) || void 0,
+                    p: location.pathname,
+                    u: location.href,
+                    s: location.search,
+                    t: document.title,
+                    r: document.referrer,
+                  })
+                }
+                n.unshift(e)
+                analytics.push(n)
+                return analytics
+              }
+            }
+            for (var n = 0; n < analytics.methods.length; n++) {
+              var key = analytics.methods[n]
+              analytics[key] = analytics.factory(key)
+            }
+            analytics.load = function (key, n) {
+              var t = document.createElement('script')
+              t.type = 'text/javascript'
+              t.async = !0
+              t.setAttribute('data-global-segment-analytics-key', i)
+              t.src =
+                'https://cdn.segment.com/analytics.js/v1/' +
+                key +
+                '/analytics.min.js'
+              var r = document.getElementsByTagName('script')[0]
+              r.parentNode.insertBefore(t, r)
+              analytics._loadOptions = n
+              analytics._writeKey = key
+            }
+            analytics.SNIPPET_VERSION = '5.2.0'
+
+            // will be used for asserting in the corresponding test
+            analytics.track('track from snippet')
```

Comment:
Shouldn't we be testing segment_analytics.track()?

Reply:
analytics here is a reference to `window['segment_analytics']`

Reply:
Right, I had some other questions, though?

Reply:
Agree with what Seth is saying - this page should just be as vanilla a snippet as can be - we can make the actual track call in the test file.

Also, does the new snippet always set `window.analytics` even if you're using a custom global key? I think that would cause a problem for users reserving `window.analytics` for something else. Wouldn't tests using this continue to pass without your a.js changes because `window.analytics` still exists with this snippet?

Reply:
> this page should just be as vanilla a snippet as can be - we can make the actual track call in the test file.

During our meeting we talked about testing if buffered calls (initiated in the snippet, before the main bundle loads) are taken care of safely - making those calls in the test file won't assert what we're trying to assert

> does the new snippet always set window.analytics even if you're using a custom global key

No, it does not. Where did you notice a read or write to `window.analytics`?

Reply:
Yes you guys are right, its not that we are testing if calls in snippet get dealt with - its the calls before load, and your proposal will cover that. Let me change that

Reply:
Done - https://github.com/segmentio/analytics-next/pull/1032/commits/fc0be16dd7d3b73ede331b75485f4c1f2327a619